File: crs-frontend/src/types/apiKey.ts
export type ApiKeyStatus = 'ACTIVE' | 'REVOKED' | 'EXPIRED'

export interface ApiKey {
  id: number
  name: string
  keyPrefix: string
  scopes: string[]
  status: ApiKeyStatus
  createdAt: string
  expiresAt: string | null
  revokedAt: string | null
  createdBy: string
}

export interface CreatedApiKeyResponse {
  id: number
  name: string
  rawKey: string
  keyPrefix: string
  scopes: string[]
  status: ApiKeyStatus
  createdAt: string
  expiresAt: string | null
  createdBy: string
}

export interface CreateApiKeyRequest {
  name: string
  scopes: string[]
  expirationDays?: number | null
}

export interface VerifyApiKeyRequest {
  apiKey: string
  requiredScope?: string
}

export interface VerifyApiKeyResponse {
  valid: boolean
  message: string
  keyName?: string
  scopes?: string[]
  hasRequiredScope: boolean
}

File: auth-service/src/main/java/com/hunre/authservice/entity/ApiKey.java
package com.hunre.authservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "api_keys")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ApiKey {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(name = "key_prefix", nullable = false, length = 30)
    private String keyPrefix;

    @Column(name = "key_hash", nullable = false, unique = true, length = 64)
    private String keyHash;

    @Column(nullable = false, length = 500)
    private String scopes;

    @Column(nullable = false, length = 20)
    private String status; // ACTIVE, REVOKED, EXPIRED

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @Column(name = "revoked_at")
    private LocalDateTime revokedAt;

    @Column(name = "created_by", length = 100)
    private String createdBy;
}
